### ISSUE TITLE
Can't activate a custom conda environment on the hub

### ISSUE THREAD

**Issue by moorepants** [LibreTexts/metalc#145]
To reproduce, open a terminal and do:

```
conda create -n cartopy-env cartopy
```

Now try:

```
conda activate cartopy-env
```

I get this:

```
jovyan@jupyter-moorepants-40gmail-2ecom:~$ conda activate cartopy-env

CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.
```

So then I call `conda init bash` and get this message:

```
jovyan@jupyter-moorepants-40gmail-2ecom:~$ conda init bash
no change     /opt/conda/condabin/conda
no change     /opt/conda/bin/conda
no change     /opt/conda/bin/conda-env
no change     /opt/conda/bin/activate
no change     /opt/conda/bin/deactivate
no change     /opt/conda/etc/profile.d/conda.sh
no change     /opt/conda/etc/conda_bash.sh
no change     /opt/conda/etc/bash_completion.d/conda
no change     /opt/conda/etc/fish/conf.d/conda.fish
no change     /opt/conda/shell/condabin/Conda.psm1
no change     /opt/conda/shell/condabin/conda-hook.ps1
no change     /opt/conda/lib/python3.6/site-packages/xontrib/conda.xsh
no change     /opt/conda/etc/profile.d/conda.csh
modified      /home/jovyan/.bashrc

==> For changes to take effect, close and re-open your current shell. <==
```

I then open a new terminal and try `conda activate cartopy-env` and get the same message:

```
jovyan@jupyter-moorepants-40gmail-2ecom:~$ conda activate cartopy-env

CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.
```

I vaguely recall activating environments working before.

**Comment by moorepants** [comment]
I also tried to open a new notebook using the cartopy-env by calling:

```
jovyan@jupyter-moorepants-40gmail-2ecom:~$ python -m ipykernel install --user --name=cartopy-env
Installed kernelspec cartopy-env in /home/jovyan/.local/share/jupyter/kernels/cartopy-env
```

Then I click "new launcher" and I see an icon for the environment in the launcher. I click the icon and an new notebook opens. When I `import cartopy` I get a `ImportError` like it isn't installed.

**Comment by moorepants** [comment]
In the notebook things are being imported from the base environment, as when I import scipy I see:

```
>>> scipy.__file__
'/opt/conda/lib/python3.6/site-packages/scipy/__init__.py'
```

**Comment by celine168** [comment]
That being said, I realize now that `nb_conda_kernels` may not have to be installed in the Default Environment itself, which would get rid of all the duplicate environments.

**Comment by celine168** [comment]
Ok so "officially", I believe this is how it works:
* you need to specify a kernel when you create the environment. This has already been specified in default-env
`conda create -n <your_env_name> <kernel> <packages>`
ex. `conda create -n r_leaflet_env r-irkernel r-leaflet`

If you want more packages in a Python environment, which seems to be considered "default" in JupyterLab ([Kernel Zero](https://github.com/jupyter/jupyter/wiki/Jupyter-kernels), according to this page), then you can also do this by just adding `nb_conda_kernels` to the list of packages without specifying your kernel.
`conda create -n <your_env_name> <packages> nb_conda_kernels`
ex. `conda create -n tensorflow_env tensorflow nb_conda_kernels`

Added new info here: https://github.com/LibreTexts/default-env/pull/38
I would like to close this issue but please check if this is working for you.

**Comment by sandertyu** [comment]
> If `conda activate` still doesn't work, try `source activate`.
> I ran into a similar problem with `conda activate` not working at first, but after running `source activate` once, all `conda activate` and `conda deactivate` commands worked.

This worked for me with the cartopy-env example;

`jovyan@jupyter-noasanders-40ucdavis-2eedu:~$ source activate cartopy-env`
`(cartopy-env) jovyan@jupyter-noasanders-40ucdavis-2eedu:~$ `

As for the most recent comment, @celine168 , what exactly would you like us to check to see if it works? I can go through and try whatever needs to be done so that this issue can be closed.

**Comment by moorepants** [comment]
We'd like for users to be able to create custom conda environments and then open up notebooks or consoles in the launcher UI for that environment. We need to figure out the exact commands they need to do so and then put it in our FAQ.

**Comment by sandertyu** [comment]
Okay, I think I've found the final piece. The instructions from Celine's merged pull request are entirely correct, although they assume an [environment directory](https://github.com/LibreTexts/default-env/pull/38/files#diff-04c6e90faac2675aa89e2176d2eec7d8L94) exists, but it does not. This directory is crucial for environment persistence. For this directory to be available, we need to create a `.condarc` file in `/home/jovyan/` and populate it as specified [here](https://zero-to-jupyterhub.readthedocs.io/en/latest/customizing/user-environment.html?highlight=repo2docker#allow-users-to-create-their-own-conda-environments-for-notebooks). Once this is done, users can create persistent conda environments and if they specify a kernel, it will show up in the list of kernals through the launcher. I've tried all of this out just by creating the necessary `.condarc` file in my `/home/jovyan/` and it seems to work every time. What needs to be done is finding a way to automatically include that `.condarc` file in every users home directory. I'm sure it can be done.
